### PR TITLE
added ability to redirect to url basing on SLO initiator referer url

### PR DESCRIPTION
**Pull request by smecnarowski**
As in topic I wasn't able to redirect to source after SLO, maybe I missed something. This is my solution for this.

### CODE REVIEW

**Comment by upwebdesign** [comment]
@smecnarowski you define the SLO url's for each of your SP's in your samlidp config. There is an example in the docs.

```php
<?php

return [
    // The URI to your login page
    'login_uri' => 'login',
    // The URI to the saml metadata file, this describes your idP
    'issuer_uri' => 'saml/metadata',
    // List of all Service Providers
    'sp' => [
        // Base64 encoded ACS URL
        'aHR0cHM6Ly9teWZhY2Vib29rd29ya3BsYWNlLmZhY2Vib29rLmNvbS93b3JrL3NhbWwucGhw' => [
            // ACS URL of the Service Provider
            'destination' => 'https://example.com/saml/acs',
            // Simple Logout URL of the Service Provider
            'logout' => 'https://example.com/saml/sls',
        ]
    ]

];
```

**Comment by smecnarowski** [comment]
this is "just" SLO url but I wanted after logging out from all service providers to redirect from IDP to destination which match to SLO initiator url

File: config/samlidp.php [hunk 1]
```diff
@@ -30,5 +30,11 @@
         //     // Your destination is the ACS URL of the Service Provider
         //     'destination' => 'https://myfacebookworkplace.facebook.com/work/saml.php'
         // ]
+    ],
+
+    // If you need to redirect after SLO depending on SLO initiator
+    // key is beginning of HTTP_REFERER value from SERVER, value is redirect path
+    'spSloRedirects' => [
```
Comment (upwebdesign) [inline config/samlidp.php]: Key names should be `snake_case`.

File: src/Http/Controllers/LogoutController.php [hunk 2]
```diff
@@ -14,6 +15,12 @@ class LogoutController extends Controller
      */
     public function index(Request $request)
     {
+        $sloRedirect = $request->session()->get('saml.sloRedirect');
```
Comment (upwebdesign) [inline src/Http/Controllers/LogoutController.php]: Variable names should be `snake_case`.

**Pull request merged**